app/src/forms/email/reminderService.js

```diff
@@ -0,0 +1,293 @@
+          periodType : forms[i].schedule.scheduleType
+        }
+      );
+    }
+
+    return reminder;
+  },
+  _getMailType : (report, reminder, late) => {
+
+    if(!reminder.enabled) return undefined;
+
+    var state = undefined;
+    const now = moment().format('YYYY-MM-DD');
+    const start_date = moment(report.dates.startDate).format('YYYY-MM-DD');
+    const end_date = (late) ? moment(report.dates.graceDate).format('YYYY-MM-DD') : moment(report.dates.closeDate).format('YYYY-MM-DD');
+    const days_diff = moment(end_date).diff(start_date,'days');
+
+    if(moment(now).isSame(start_date)) {
+      return EmailTypes.REMINDER_FORM_OPEN;
+    }
+
+    if(report.dates.closeDate==null || days_diff<=3 ) return  state;
+
+    if(service.checkIfInMiddleOfThePeriod(reminder.intervalType, now, start_date, days_diff )){
+      return EmailTypes.REMINDER_FORM_NOT_FILL;
+    }
+
+    const yend_date = moment(end_date).subtract(1, 'day');
+
+    if(moment(now).isSame(yend_date)){
+      return EmailTypes.REMINDER_FORM_WILL_CLOSE;
+    }
+
+    return state;
+  },
+  checkIfInMiddleOfThePeriod : (type, now, start_date,  days_diff )=>{
+    if (days_diff < 6 ) return false;
+
+    if(type!=null && type) {
+      for (const key in periodType) {
+        const interval = moment(now).diff(start_date, periodType[key].regex);
+        if( key==type &&  interval%periodType[key].value==0 ) {
+          return true;
+        }
+      }
+    } else {
+      let interval = Math.floor(days_diff/2);
+      // eslint-disable-next-line no-console
+      let mail_date = moment(start_date).add(interval, 'days').format('YYYY-MM-DD');
+      return moment(now).isSame(mail_date);
+    }
+
+    return false;
+  },
+  _runQueries : async (queries) => {
+    var obj = queries[2];
+    obj.fillers = [];
+    await queries[0].then(function(data) {
+      obj.submiters = data;
+    });
+    if(obj.submiters && obj.submiters.length>0){
+      await queries[1].then(function(data2) {
+        obj.fillers = (data2) ? data2 : [];
+      });
+    }
+    return  obj;
+  },
+  getDifference : async (array1, array2) => {
+    return array1.filter(object1 => {
+      return !array2.some(object2 => {
+        return object1.userId === object2.userId;
+      });
+    });
+  },
+  _initStatement : async (obj) => {
+    const statement = {};
+    statement.form = obj.form;
+    statement.report = obj.report;
+    statement.state = obj.state;
+    statement.submiters = [];
+
+    if(!obj.submiters || obj.submiters.length == 0) return statement;
+
+    if(statement.state != EmailTypes.REMINDER_FORM_OPEN ){
+      if(obj.fillers &&  obj.fillers.length != 0 ) {
+        statement.submiters = await service.getDifference(obj.submiters, obj.fillers);
+      } else {
+        statement.submiters = obj.submiters;
+      }
+
+    } else {
+      statement.submiters = obj.submiters;
+    }
+    return statement;
+  },
+  _getReferer : () => {
```

Comment:
This whole function shouldn't even exist, you should be getting the referer in the same way the emailService gets the referer.